File: architecture/evidence/issues/153/offline-profile/review-probes/independent-deep-probe.mjs
import {readFileSync,writeFileSync} from 'node:fs';
import {buildAnnualAccountsPayload} from './apps/web/app/lib/annual-accounts.ts';
import {renderAnnualAccountsXml} from './apps/web/app/lib/annual-accounts-xml.ts';
const c=JSON.parse(readFileSync('./architecture/evidence/issues/153/characterization/legacy-pure-characterization.json'));
const results=[];
for(const depth of [100,600,1200]){const input=structuredClone(c.payloadCases[0].input);let v='ignored';for(let i=0;i<depth;i++)v=[v];input.annualData.ignored=v;try{results.push({depth,status:'ok',fieldCount:buildAnnualAccountsPayload(input).fields.length})}catch(e){results.push({depth,status:e.name})}}
writeFileSync(process.argv[2],JSON.stringify(results,null,2)+'\n');console.log(results);

File: architecture/evidence/issues/153/offline-profile/review-probes/independent_deep_probe.py
import json
from pathlib import Path
from talli_backend.modules.annual_accounts_filing.public import AnnualAccountsSource,build_annual_accounts
c=json.loads(Path('architecture/evidence/issues/153/characterization/legacy-pure-characterization.json').read_text())
results=[]
for depth in (100,600,1200):
 value=json.loads(json.dumps(c['payloadCases'][0]['input'])); extra='ignored'
 for _ in range(depth):extra=[extra]
 value['annualData']['ignored']=extra
 try:
  result=build_annual_accounts(AnnualAccountsSource(value['incomeYear'],value['annualData'],tuple(value['ledgerEntries'])))
  results.append(dict(depth=depth,status='ok',fieldCount=len(result.fields)))
 except Exception as error:results.append(dict(depth=depth,status=type(error).__name__,message=str(error)))
print(json.dumps(results,indent=2))
